sklearn/utils/estimator_checks.py

```diff
@@ -589,15 +592,30 @@ def check_fit2d_1sample(name, estimator_orig):
         estimator.n_clusters = 1
 
     set_random_state(estimator, 1)
+
+    msgs = ["Found array with 1 sample(s)",
+            "number of classes has to be greater than one",
+            "greater than the number of samples: 1",
+            "requires 2 classes",
+            "requires 2 or more distinct classes",
+            "n_samples = 1",
+            "n_samples=1",
+            "one sample",
+            "one class",
+            "less than 2 classes",
+            ]
+
     try:
         estimator.fit(X, y)
-    except ValueError:
-        pass
+    except ValueError as e:
+        if all(msg not in repr(e) for msg in msgs):
```

Comment:
This can be done with `assert_raises_regexp` FWIW.

But this is quite a rag bag of error messages. I guess it's an improvement. But do we intend to cut down the number of acceptable ways for an estimator to state this? If so, maybe we should comment in here what the preferred message is.

Reply:
We are testing that `fit` either works or returns an informative message. Can we really use `assert_raises_regexp` here?

Reply:
I agree with @albertcthomas I don't think that assert_raises_regexp works here

Reply:
okay